_ml-commons-plugin/api.md

```diff
@@ -0,0 +1,968 @@
+---
+layout: default
+title: API
+has_children: false
+nav_order: 99
+---
+
+# ML Commons API 
+
+---
+
+#### Table of contents
+- TOC
+{:toc}
+
+
+---
+
+The Machine Learning (ML) commons API lets you create, train, and store machine learning algorithms both synchronously and asynchronously.  
+
+In order to train tasks through the API, three inputs are required. 
+
+- Algorithm name: Usually `FunctionaName`. This determines what algorithm the ML Engine runs.
+- Model hyper parameters: Adjust these parameters to make the model train better.  You can also implement `MLAgoParamas` to build custom parameters for each model.
+- Input data: The data input that teaches the ML model. To input data, query against your index or use data frame.
+
+## Train model
+
+Training can occur both synchronously and asynchronously.
+
+### Request 
+
+The following examples use the kmeans algorithm to train index data.
+
+**Train with kmeans synchronously** 
+
+```json
+POST /_plugins/_ml/_train/kmeans
+{
+    "parameters": {
+        "centroids": 3,
+        "iterations": 10,
+        "distance_type": "COSINE"
+    },
+    "input_query": {
+        "_source": ["petal_length_in_cm", "petal_width_in_cm"],
+        "size": 10000
+    },
+    "input_index": [
+        "iris_data"
+    ]
+}
+```
+
+**Train with kmeans asynchronously**
+
+```json
+POST /_plugins/_ml/_train/kmeans?async=true
+{
+    "parameters": {
+        "centroids": 3,
+        "iterations": 10,
+        "distance_type": "COSINE"
+    },
+    "input_query": {
+        "_source": ["petal_length_in_cm", "petal_width_in_cm"],
+        "size": 10000
+    },
+    "input_index": [
+        "iris_data"
+    ]
+}
+```
+
+### Response
+
+**Synchronously**
+
+For synchronous responses, the API returns the model_id, which can be used to get info or modify the model.
```

Comment:
should we point it out that this model_id is a required input for predict API?

Reply:
Yes.

Reply:
Added a note to the predict APi